### DISCUSSION TITLE
mjpeg cam support [Config Support]:

### DISCUSSION THREAD

**Discussion by JakeLC15** [blakeblackshear/frigate#13031] [Config Support]
### Describe the problem you are having

Sharing this config to possibly update the config files

### Version

0.14.0-da913d8

### Frigate config file

```yaml
mqtt: enabled: true host: core-mosquitto port: 1883 user: **** password: **** topic_prefix: frigate detectors: coral: type: edgetpu device: usb ffmpeg: #hwaccel_args: # - -c:v:1 # - h264_v4l2m2m global_args: - -hide_banner - -loglevel - info go2rtc: streams: camera_2: - http://192.168.0.134:8080 yi_1: - rtsp://192.168.0.160/ch0_0.h264 - ffmpeg:yi_1#audio=aac yi_1_sub: - rtsp://192.168.0.160/ch0_1.h264 cameras: camera_1: enabled: false ffmpeg: inputs: - path: http://192.168.0.119:8080 # http://127.0.0.1:1984/api/stream.mjpeg?src=camera_1 roles: - detect - record input_args: -avoid_negative_ts make_zero -fflags nobuffer -flags low_delay -strict experimental -fflags +genpts+discardcorrupt -use_wallclock_as_timestamps 1 -c:v mjpeg output_args: record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v libx264 -an detect: #width: 800 #height: 600 fps: 3 objects: track: - person - dog motion: # threshold: 200 improve_contrast: true record: enabled: false retain: days: 1 events: retain: # mode: motion default: 5 camera_2: enabled: true ffmpeg: hwaccel_args: - -c:v:1 - h264_v4l2m2m inputs: - path: http://127.0.0.1:1984/api/stream.mjpeg?src=camera_2 roles: - detect - record input_args: -avoid_negative_ts make_zero -fflags nobuffer -flags low_delay -timeout '10000000' -strict experimental -fflags +genpts+discardcorrupt -use_wallclock_as_timestamps '1' -c:v mjpeg output_args: record: preset-record-generic #-f segment -segment_time 10 -segment_format mp4 -g 10 -reset_timestamps # 1 -strftime 1 -c:v libx264 -an # rtmp: -c:v libx264 -an -f flv detect: width: 800 height: 600 fps: 5 objects: track: - person - dog - car motion: # threshold: 15 improve_contrast: true record: enabled: true retain: days: 1 mode: active_objects events: retain: default: 5 snapshots: enabled: true timestamp: true retain: default: 10 objects: person: 15 car: 15 dog: 5 live: stream_name: camera_2 # zones: # zone_0: # coordinates: 800,0,586,23,507,227,266,225,0,0,0,600,800,600 # review: # alerts: # required_zones: # - zone_0 tablet: enabled: true ffmpeg: inputs: - path: http://192.168.0.140:8080/video roles: - detect - record input_args: -avoid_negative_ts make_zero -fflags nobuffer -flags low_delay -strict experimental -fflags +genpts+discardcorrupt -use_wallclock_as_timestamps 1 -c:v mjpeg output_args: record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v libx264 -an # rtmp: -c:v libx264 -an -f flv detect: #width: 800 #height: 600 fps: 3 objects: track: - person record: enabled: false retain: days: 0 events: retain: default: 5 motion: mask: - 459,245,641,216,664,375,498,395 yi_1: enabled: true ffmpeg: output_args: record: preset-record-generic-audio-aac inputs: - path: rtsp://127.0.0.1:8554/yi_1?video&audio #rtsp://192.168.0.160/ch0_0.h264 input_args: preset-rtsp-restream roles: - record - path: rtsp://127.0.0.1:8554/yi_1_sub input_args: preset-rtsp-restream roles: - detect detect: fps: 5 objects: track: - person - dog record: enabled: true retain: days: 0 events: retain: default: 5 snapshots: enabled: true live: stream_name: yi_1 motion: mask: 0.541,0.04,0.55,0.208,0.719,0.209,0.718,0.055 version: 0.14
```

### Relevant log output

```shell
2024-08-13 15:16:07.198446623 [INFO] Preparing new go2rtc config... 2024-08-13 15:16:07.334129456 [INFO] Got IP address from supervisor: 192.168.0.127 2024-08-13 15:16:07.475026149 [INFO] Got WebRTC port from supervisor: 8555 2024-08-13 15:16:08.754799938 [INFO] Starting go2rtc... 2024-08-13 15:16:08.999207946 11:16:08.996 INF go2rtc platform=linux/arm64 revision=b2399f3 version=1.9.2 2024-08-13 15:16:08.999222243 11:16:08.996 INF config path=/dev/shm/go2rtc.yaml 2024-08-13 15:16:08.999225539 11:16:08.998 INF [api] listen addr=:1984 2024-08-13 15:16:08.999229039 11:16:08.998 INF [rtsp] listen addr=:8554 2024-08-13 15:16:08.999732181 11:16:08.999 INF [webrtc] listen addr=:8555/tcp 2024-08-13 15:16:16.926439952 [INFO] Starting go2rtc healthcheck service... 2024-08-13 15:23:01.183593758 11:23:01.182 WRN [rtsp] error="streams: codecs not matched: JPEG => H264, H265, AAC" stream=camera_2 2024-08-13 15:30:41.192784664 11:30:41.192 WRN [rtsp] error="streams: codecs not matched: JPEG => H264, H265, AAC" stream=camera_2
```

### Operating system

HassOS

### Install method

HassOS Addon

### Object Detector

Coral

### Any other information that may be helpful

Camera labeled camera_2 is working perfectly with less than 10% ffmpeg cpu usage. It records and detects without issue. This is a esp32 cam which only outputs mjpeg.
The mjpeg is passed through go2rtc and the stream is not recoded to h264. Frigate uses the stream and saves as mp4 file.
It seems that the last update (0.14) is able to handle the mjpeg stream.
It did break the ability to see events with the record: false. Snapshots are not shown in the timeline if record is off regardless of snapshot config. Snapshots are saved to file. Detection works and saves images. None are shown in timeline with record set to false. I believe this related to mjpeg.
There is an error log of codecs not matched. Which is correct. It is looking for h264 but received mjpeg. But it processed it anyway!

Also live view usually fails to load. Again an mjpeg related side effect from a $5 camera

Related esphome config
'''
esp32_camera:
  external_clock:
    pin: GPIO0
    frequency: 20MHz
  i2c_pins:
    sda: GPIO26
    scl: GPIO27
  data_pins: [GPIO5, GPIO18, GPIO19, GPIO21, GPIO36, GPIO39, GPIO34, GPIO35]
  vsync_pin: GPIO25
  href_pin: GPIO23
  pixel_clock_pin: GPIO22
  power_down_pin: GPIO32
  #vertical_flip: false
  #horizontal_mirror: false

  # Image settings
  name: Cam2
  icon: mdi:camera
  #idle_framerate: 0.2 fps
  max_framerate: 5 fps
  resolution: 800x600
  jpeg_quality: 10
  brightness: 2
  #aec_mode: manual
  agc_mode: manual
  #wb_mode: home
  #aec_value: 500
  agc_gain_ceiling: 2x
  
esp32_camera_web_server: 
  - port: 8080
    mode: stream 
#  - port: 8081 
#    mode: snapshot

switch:
  - platform: restart 
    name: cam2_restart
  # ...
output:
    #flashlight 
  - platform: gpio 
    pin: GPIO4 
    id: gpio_4
light:
    #flashlight 
  - platform: binary 
    output: gpio_4 
    name: light
'''

**Comment by NickM-27** [comment]
the suggestion is to configure go2rtc to do the transcoding

**Reply by JakeLC15** [reply]
I am happy with it passing the mjpeg through. It uses less cpu resources. The lowest I could get the h264 transcoding was about 40% per camera. Not enough cpu with a rpi4! I'm guessing that the frigate media player doesn't work well with mjpeg. Or it is hass os that doesn't support the codec. Frigate video that is saved only seems to play on vlc.

**Reply by NickM-27** [reply]
in order to play back recordings, using h264 or h265 is a requirement
